proposed/standard-ci-env-variables.md

```diff
@@ -0,0 +1,48 @@
+# Standardized Environment Variables for CI Services
+
+Many CI services set environment variables that can be used by developer builds. Many of the environment variables are common in scope and theme across CI services, however, the specific variable names are different. We need a set of standardized environment variables across CI services so that an application platform like .NET can rely on them, and provide a common user experience across CI services.
+
+## Context
+
+The .NET Core SDK is implementing new scenarios that require source control manager information. An initial idea was to call out to specific source control tools to get this information. This approach is problematic since it requires an implementation for each source control manager and it has the potential to be fragile and slow.
+
+It turns out that CI services provide all of the information that these new .NET Core scenarios need via environment variables. Also, there is a strong relationship between these scenarios and official CI-provided builds (much less on local developer builds). As a result, relying on the CI-provided environment variables is attractive.
+
+To make it possible to provide application platform provided experiences across CI services, we need a standardized set of environment variables that are supported across those same CI services. Ideally, this set of environment variables would be supported across multiple CI services and useful for multiple application environments, not just .NET.
+
+## Proposed environment variables
+
+The .NET Core SDK needs are oriented around source control. As a result, the intial list is source control oriented, but there is no affinity to source control on the general idea of standardized environment variables.
+
+It is important that these environment variables do not conflict with other variables. To avoid that, all environment variables will be prepended with "STANDARDCI\_". This name is a first proposal for the idea and it may get changed based on feedback.
+
+* **STANDARDCI\_REPOSITORYCOMMITID** -- Commit hash / ID; Example: 2ba93796dcf132de447886d4d634414ee8cb069d
+* **STANDARDCI\_REPOSITORYROOT** -- Root of repository; Example: D:\repos\corefx
+* **STANDARDCI\_REPOSITORYNAME** -- Name of repository; Example: dotnet\corefx
+* **STANDARDCI\_REPOSITORYURI** -- Uri for repository; Example: https://github.com/dotnet/corefx
```

Comment:
Yes, I think that if this is going to be an all-CI standard, then it _must_ clearly state what happens to this variables if there is more then one repository involved in the CI build (not a rare thing at all in our world BTW).

Reply:
That's a great question. Two thoughts ...

* Do single binaries get built from multiple repos?
* How do the existing CI environment variables behave in that scenario?

Reply:
@richlander: 
* Can't rule out for all cases. In our scenarios, _often_ you might choose one primary repository which is mostly responsible for making the output binary. Anyways, there're no CI settings for this choosing, it goes untold.
* Considering TeamCity, 
  * `build.vcs.number` is only present if there is one source control attached to the build. Otherwise there would be multiple variables called `build.vcs.number.id-of-source-control-root-here` (each source control root which you attach to this build has an ID in TeamCity), one per root, with their respective revision numbers.
  * `teamcity.build.checkoutDir` isn't the repository root at all, actually. It's just the working root of the build run. When attaching a source control root, you specify the folder (or folders) where its files go. Even in case of one single root that does not have to be the working directory root. I'm not sure if you can get the actual repository root folders at all in TC props.